addons/udes_stock/models/stock_picking_batch.py

```diff
@@ -231,51 +416,24 @@ def _create_batch(self, user_id, picking_type_id, picking_priorities=None):
 
         batch = PickingBatch.create({'user_id': user_id})
         picking.batch_id = batch.id
-        batch.write({'state': 'in_progress',
-                     'u_ephemeral': True})
+        batch.write({'u_ephemeral': True})
+        batch.confirm_picking()
 
         return batch
 
-    def _check_batches(self, user_id, batches=None, raise_error=True):
-        """
-        Checks if batches are complete, and marks as done.
-
-        If no batch is provided, searches all batches in progress
-        for the specified user.
-
-        Optionally raises a ValidationError if any non-draft picking
-        is incomplete.
-        """
-        Picking = self.env['stock.picking']
-        PickingBatch = self.env['stock.picking.batch']
-
-        if not batches:
-            batches = PickingBatch.search([('user_id', '=', user_id),
-                                           ('state', '=', 'in_progress')])
-
+    def _check_user_batch_in_progress(self, user_id=None):
```

Comment:
Sorry to repeat, but the optional arg requires reading `get_user_batches` to understand what's going on. I suggest to only use optional args when necessary.